src/core/public/application/types.ts

```diff
@@ -802,8 +818,13 @@ export interface ApplicationStart {
    *
    * @param url - an absolute URL, an absolute path or a relative path, to navigate to.
    */
-  navigateToUrl(url: string): Promise<void>;
-
+  navigateToUrl(url: string, options?: NavigateToUrlOptions): Promise<void>;
+  /**
+   * Navigate to a given url
+   *
+   * The method removes the eventListener and assigns the window location to the url provided
+   */
+  navigateToUrlSkipUnload(url: string): void;
```

Comment:
The naming here implies a very specific scope for this function: all it is really doing is saving the user from configuring `navigateToUrl` with the new `skipAppLeave` option.

Based on the discussion in the original issue about having a dedicated API for navigating to a url unconditionally, I wonder if we shouldn't go with something broader like `forceNavigateToUrl` (or similar)

That way we aren't tying this to a specific piece of functionality in case we add other "pre-navigation" checks in the future. WDYT?

Reply:
Naming the navigation method `forceNativateToUrl` was my first instinct but then I changed my mind to be more descriptive about what the method does.
Naming is hard though, so maybe your option works too.

Reply:
I think it comes down to what we see the scope of this API being moving forward. If the purpose of the API is to allow navigating to a URL while skipping any events triggered by the `beforeonload` event listener, then I think the current naming makes sense. If the purpose is to allow navigating to a URL without any additional checks being performed (not necessarily specific to `beforeonload`), then I'd vote for something more generic like forceNavigateToUrl or similar

Also, thinking about it more, do we need the `skipAppLeave` option at all if we have a purpose-built API like this?

Reply:
> scope of this API being moving forward

The intention of the additional navigation method is to have a hook for security to plug into when dealing with session timeouts that force a user to log in again. As an aside: we don't want to encourage navigation that arbitrarily skips the additional checks applications have before unloading. If we start arbitrarily skipping application cleanup behavior we'll probably end up with unintentional side effects, considering how much state gets passed around at times.

> do we need the skipAppLeave option at all

The flags allow plugin authors to specifically choose the behavior they want rather than having to change the application navigation method itself. The new method doesn't handle `deepLinks` or `state` which is taken care of by the other nav methods.

Reply:
Update: we've decided to take a different approach and won't be adding a new API anymore. See https://github.com/elastic/kibana/pull/128671#discussion_r838431305